PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift

```diff
@@ -100,19 +101,18 @@ struct DiningVenueDetailMenuView: View {
                 }
             }
         }
-        
-        .onChange(of: currentMenu) { _ in
-            print((currentMenu?.service ?? "no menu") + " on " + menuDate.description)
-            selectedStation = currentMenu?.stations.first ?? nil
-        }
-        .onChange(of: menuDate) { newDate in
+        .onChange(of: menuDate) { _ in
             Task.init() {
-                await diningVM.refreshMenus(cache: true, at: newDate)
-                menuDate = newDate
+                await diningVM.refreshMenus(cache: true, at: menuDate)
                 menus = diningVM.diningMenus[venue.id]?.menus ?? []
                 currentMenu = getMenu()
             }
         }
         
+        .onChange(of: currentMenu) { _ in
+            print((currentMenu?.service ?? "no menu") + " on " + menuDate.description)
+            selectedStation = nil
```

Comment:
Just wondering why we're setting this to nil?

Reply:
SelectedStation is connected to the scrollviewproxy, so changing it causes an immediate scroll, whereas setting it to nil allows this value to be automatically adjusted by the user scrolling.

https://github.com/user-attachments/assets/575e0870-c2c7-4590-b5a8-f2cd1d1671a5